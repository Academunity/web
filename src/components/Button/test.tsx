import { render, screen } from 'utils/tests';
import { Email } from 'styled-icons/material-outlined';

import { Button } from '.';

describe('<Button />', () => {
  it('should render the medium size by default', () => {
    const { container } = render(<Button>Sign in</Button>);

    expect(screen.getByRole('button', { name: /Sign in/i })).toHaveStyle({
      height: '4rem',
      'font-size': '1.4rem',
      padding: '0.8rem 3.2rem',
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the small size', () => {
    render(<Button size="small">Sign in</Button>);

    expect(screen.getByRole('button', { name: /Sign in/i })).toHaveStyle({
      height: '3rem',
      'font-size': '1.2rem',
    });
  });

  it('should render the large size', () => {
    render(<Button size="large">Sign in</Button>);

    expect(screen.getByRole('button', { name: /Sign in/i })).toHaveStyle({
      height: '5rem',
      'font-size': '1.6rem',
      padding: '0.8rem 4.5rem',
    });
  });

  it('should render a full width version', () => {
    render(<Button fullWidth>Sign in</Button>);

    expect(screen.getByRole('button', { name: /Sign in/i })).toHaveStyle({
      width: '100%',
    });
  });

  it('should render an icon version', () => {
    render(<Button icon={<Email data-testid="icon" />}>Sign in</Button>);

    expect(screen.getByText(/Sign in/i)).toBeInTheDocument();
    expect(screen.getByTestId(/icon/i)).toBeInTheDocument();
  });

  it('should render a minimal version', () => {
    render(
      <Button icon={<Email data-testid="icon" />} minimal>
        Sign in
      </Button>,
    );

    expect(screen.getByRole('button', { name: /Sign in/i })).toHaveStyle({
      background: 'none',
      color: '#BA75FF',
    });

    expect(screen.getByRole('button', { name: /Sign in/i })).toHaveStyleRule(
      'background',
      'none',
      {
        modifier: ':hover',
      },
    );
  });

  it('should render a disabled button', () => {
    render(<Button disabled>Sign in</Button>);

    expect(screen.getByRole('button', { name: /Sign in/i })).toHaveStyleRule(
      'cursor',
      'not-allowed',
      {
        modifier: ':disabled',
      },
    );
  });

  it('should render Button as a link', () => {
    render(
      <Button as="a" href="/link">
        Sign in
      </Button>,
    );

    expect(screen.getByRole('link', { name: /Sign in/i })).toHaveAttribute(
      'href',
      '/link',
    );
  });
});
