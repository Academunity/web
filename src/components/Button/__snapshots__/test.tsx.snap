// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the medium size by default 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #BA75FF;
  color: rgba(255,255,255,0.87);
  border-radius: 0.4rem;
  border: none;
  padding: 1.5rem 3.2rem;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-transition: background-color 0.2s ease-in-out;
  transition: background-color 0.2s ease-in-out;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c0:hover {
  background: #b266ff;
}

<button
  class="c0"
>
  <span>
    Sign in
  </span>
</button>
`;
